oteljava/metrics/src/main/scala/org/typelevel/otel4s/oteljava/metrics/CounterBuilderImpl.scala

```diff
@@ -125,7 +128,7 @@ private[oteljava] object CounterBuilderImpl {
           val counter = builder.ofDoubles().build()
 
           val backend = new Counter.Backend[F, A] {
-            val meta: InstrumentMeta[F] = InstrumentMeta.enabled
+            def meta: InstrumentMeta.Dynamic[F] = instrumentMeta
```

Comment:
I think this should still be a `val`; otherwise it has to pull `instrumentMeta` into the "closure" anyway

Reply:
Fair enough. Fixed.